### PR TITLE
Move box shadow building code to a separate source file.

### PR DESCRIPTION
**Pull request by glennw**
This is prep work for some of the upcoming box shadow optimizations. It makes sense to move this into a separate file since there will be quite a lot of box shadow specific code.

This PR itself has no functional changes. It just moves code around.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/1937)

### CODE REVIEW

File: webrender/src/box_shadow.rs [hunk 1]
```diff
@@ -0,0 +1,291 @@
+use api::{ColorF, LayerPoint, LayerRect, LayerSize, LayerVector2D};
```
Comment (nical) [inline webrender/src/box_shadow.rs]: Don't we need a license header at the top of this file?

Comment (glennw) [reply webrender/src/box_shadow.rs]: Oops, fixed.

**Pull request merged**